### PR TITLE
Allow objects to be given as macro values in RTC config

### PR DESCRIPTION
**Pull request by bradfrizzell**
Simply stringify any objects. 

See https://github.com/ampproject/amphtml/issues/12801

### CODE REVIEW

File: examples/doubleclick-rtc.amp.html [hunk 1]
```diff
@@ -30,5 +30,17 @@ <h2>Doubleclick with RTC</h2>
             rtc-config='{"urls": ["https://www.example.biz/?pvi=PAGEVIEWID&href=HREF&ds=ATTR(data-slot)&h=ATTR(height)&w=ATTR(width)&ovw=ATTR(data-override-width)&json=ATTR(data-json)&ovh=ATTR(data-override-height)"], "timeoutMillis": 500}'
             json='{"targeting":{"food":["burgers","chicken"]},"categoryExclusions":["health"],"tagForChildDirectedTreatment":0}'>
     </amp-ad>
+
+    <amp-ad width="320" height="50"
+            type="doubleclick"
+            data-slot="/4119129/mobile_ad_banner"
+            rtc-config='{
```
Comment (jonkeller) [inline examples/doubleclick-rtc.amp.html]: Nit: could just make all rtc-config be on one line, like on line 30 above. Or could indent the JSON object (e.g. 2 more spaces on line 38, 4 more on line 39, etc. Or not, up to you.

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 2]
```diff
@@ -283,6 +283,15 @@ export function validateRtcConfig_(element) {
     return null;
   }
 
+  const vendors = rtcConfig['vendors'] || {};
+  Object.keys(vendors).forEach(vendor => {
+    Object.keys(vendors[vendor]).forEach(key => {
+      if (isObject(vendors[vendor][key])) {
+        vendors[vendor][key] = JSON.stringify(vendors[vendor][key]);
```
Comment (jonkeller) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: Maybe a comment explaining why this is being done?

File: examples/doubleclick-rtc.amp.html (hunk 1 above)
Comment (bradfrizzell) [reply examples/doubleclick-rtc.amp.html]: Going to leave as is for my own readability. And I think I am the only one who ever looks at this file anyways :D

File: extensions/amp-a4a/0.1/real-time-config-manager.js (hunk 2 above)
Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/real-time-config-manager.js]: Will do

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 3]
```diff
@@ -283,6 +283,21 @@ export function validateRtcConfig_(element) {
     return null;
   }
 
+  // The following loop is to allow vendor macros to be set as objects.
```
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: We should do this at the time we build the macros. Instead what if we modify above macro setup as follows:

validVendorMacros[macro] = rtcConfig['vendors'][vendor][macro];	

becomes:
	         
validVendorMacros[macro] = (isObject(rtcConfig['vendors'][vendor][macro]) ? JSON.stringify : val => val).call(null, rtcConfig['vendors'][vendor][macro]);

This saves a later, redundant iteration

**Pull request merged**